mip/solution_format.go

```diff
@@ -0,0 +1,63 @@
+package mip
+
+import (
+	"github.com/nextmv-io/sdk/run/schema"
+	"github.com/nextmv-io/sdk/run/statistics"
+)
+
+// CustomResult is an example of custom statistics that can be added to the
+// output and used in experiments.
+type CustomResult struct {
+	Columns int    `json:"columns,omitempty"`
+	Rows    int    `json:"rows,omitempty"`
+	Status  string `json:"status,omitempty"`
+}
+
+// Format the MIP solution into the output format that the runner expects. The
+// solution represents the decisions that were made by the solver, translated
+// into the domain of the problem.
+func Format(
+	options any,
+	solution any,
+	solverSolution Solution,
+) schema.Output {
+	output := schema.NewOutput(options, []any{})
+	output.Statistics = statistics.NewStatistics()
+	output.Statistics.Run = &statistics.Run{}
+	output.Statistics.Result = &statistics.Result{}
+	if solverSolution == nil || !solverSolution.HasValues() || solverSolution.IsInfeasible() {
+		return output
+	}
+
+	duration := solverSolution.RunTime().Seconds()
+	output.Statistics.Result.Duration = &duration
+	output.Statistics.Run.Duration = &duration
+
+	value := statistics.Float64(solverSolution.ObjectiveValue())
+	output.Statistics.Result.Value = &value
+
+	output.Solutions = []any{solution}
+
+	return output
+}
+
+// DefaultStatistics creates default custom statistics for a given solution.
+func DefaultStatistics(model Model, solution Solution) CustomResult {
```

Comment:
I find the naming here a bit confusing. Can't we rather just add these statistics on the format itself? I would like to always have them, I think. 
If not, I would prefer something like this, I think:
```go
func AddExtendedStatistics(model Model, solution Solution, output schema.Output) {}
```
Or something like that. 
Alternatively, we can rename this to `mip.DefaultStatisticsResult` or similar. 
The reason I am a bit confused by it is that the Statistics (and DefaultStatistics suggest that) is more than the result part. There is also the progression part which we typically add in nextroute for example.

Let me know, if I don't make any sense. :sweat_smile:

Reply:
The reason why I added the statistics as a separate function is to make it super clear in the template that there are these things called statistics. If it is buried inside the `Format` function, then it all happens automagically, so I wanted to shed visibility there. I think it is nicer for people to simply plug out the default function and use their own function to apply stats to the output. That is why it is separate.

On another note, I'll change the naming. Thanks.